### ISSUE TITLE
Availability of the checkpoints

### ISSUE THREAD

**Issue by lamthuy** [mdsunivie/deeperwin#12]
Hi,
I looked at your paper Towards a Foundation Model for Neural Network Wavefunctions. I found it interesting and would like to try the pretrained models. I wonder if the checkpoints used in the paper are available? 
Also in the paper how long does it take to train your models on the 18 compounds datasets?
In the paper, the models were pretrained with 500k steps with the dataset of 360 geometries. Does 1 pretraining step mean 1 gradient update step using 1 geometry?
Thank you!

**Comment by LeonGerBerlin** [comment]
Hi, 
thanks for your interest in our work! 

1. Yes, the checkpoints are available on figshare:
https://figshare.com/articles/online_resource/Pre-trained_neural_wavefunction_checkpoints_for_the_GitHub_codebase_DeepErwin/23585358/1

For the "Towards a Foundation Model for Neural Network Wavefunctions" you would need to take the checkpoint: 
tinymol_v10_2023-02-26_18x20_500k.zip

2. Regarding runtime, it was roughly 4-5 days on 4 A100 GPUs. 
3. Yes, one pretraining step (= one gradient update step) contains only one geometry. 
Let me know if you encounter some issue loading the checkpoints. 

Best, 
Leon

**Comment by lamthuy** [comment]
Hi Leon,
Thank you for detailed answer. 
Regarding my question above:
```
In the paper, the models were pretrained with 500k steps with the dataset of 360 geometries. Does 1 pretraining step mean 1 gradient update step using 1 geometry?
```
With 500k pretraining steps on 360 geometries, does it mean that for each geometry only perform roughly 500000/360  ~ 1388 gradient updates or each geometry is updated 500K times?

I have  more questions regarding the fine-tuning in the paper (Figure 5),
1. whether fine-tuning was done per geometry in the two test sets (small in-distribution and large out of distribution), or combining all geometries and do fine-tuning at the same time as did with pretraining?
2. if fine-tuning was done all together with all geometries, whether 32k steps in that figure still means 32k gradient update steps overall or per geometry we have 32k updates?
Thank you for your help!
Lam

**Comment by LeonGerBerlin** [comment]
Hi, 

> In the paper, the models were pretrained with 500k steps with the dataset of 360 geometries. Does 1 pretraining step mean 1 gradient update step using 1 geometry?

Yes, as written above, 1 pretraining steps means 1 gradient update steps using 1 geometry. 

> With 500k pretraining steps on 360 geometries, does it mean that for each geometry only perform roughly 500000/360 ~ 1388 gradient updates or each geometry is updated 500K times?

We use 500k steps in total across all geometries. Meaning we have roughly ~1388 gradient updates per geometry. In general we choose the geometry based on the std. dev. of the local energy so some geometries have more update steps and others less. 

> Whether fine-tuning was done per geometry in the two test sets (small in-distribution and large out of distribution), or combining all geometries and do fine-tuning at the same time as did with pretraining?

We considered in this case each molecule separately (3 compounds for small in-distribution and 4 compounds for the large out-of-distribution dataset). Each of the 7 molecules had 10 geometries and during fine-tuning we only shared weights across the 10 geometries of one molecule. In other words we didn't share weights across compounds in the fine-tuning stage.  

> If fine-tuning was done all together with all geometries, whether 32k steps in that figure still means 32k gradient update steps overall or per geometry we have 32k updates?

In the plot we show the total number of steps. Since in Fig. 5 we shared the weights across 10 geometries, so 3.2k gradient update steps per geometry.

Thanks again for your interest in our work. Let me know if still something is unclear. 

Best regards,

Leon

**Comment by lamthuy** [comment]
That's a great detail about the experimental settings. In the paper on page 18, the paper stated:
```
E_corr := E_HF - E_CCSD(T) 
```
Whether the correct formula should be?
```
E_corr := E_CCSD(T) -  E_HF 
```

**Comment by MScherbela** [comment]
Yes, you are correct, this is indeed a typo.
While it's in general irrelevant whether you define the correlation energy to be positive or negative, the equation 2 and 3 on p. 18 are currently inconsistent. We'll correct it in a revised version!

Luckily this is only a typo and should not affect the energies in our experiments.

**Comment by lamthuy** [comment]
Thank you both! I think it is a small typo.
